packages/@ember/debug/data-adapter.ts

```diff
@@ -34,13 +35,27 @@ type WrappedRecord<T> = {
 
 type RecordCallback<T> = (records: Array<{ columnValues: object; object: T }>) => void;
 
+// Represents the base contract for iterables as understood in the GLimmer VM
+// historically. This is *not* the public API for it, because there *is* no
+// public API for it. Recent versions of Glimmer simply use `Symbol.iterator`,
+// but some older consumers still use this basic shape.
+interface GlimmerIterable<T> {
+  length: number;
+  forEach(fn: (value: T) => void): void;
```

Comment:
Aside: I spent some time mucking with `.forEach()` vs. `Symbol.iterator` [on perf.link][p] and in microbenchmarks, preferring `.forEach()` seems to be a good win in most scenarios:

- `.forEach()` consistently wins by at least 65% in JSC
- `.forEach()` *always* wins by at least a factor of 5 in SpiderMonkey
- `.forEach()` is around 100% faster for small-ish arrays and never less than 20% slower for even very large arrays (I tested with up to 1 million items)

These are microbenchmarks, so they may be fairly misleading in real-world scenarios, but we should think about whether it might make sense to get some real-world data on that impact and if so consider whether/how to employ it in other places in the app. Here, I suggest we follow up on this PR by just using `.forEach()`: it lets us get rid of these TS shenanigans entirely, and for the places this is used I *suspect* that just using `.forEach()` will always be a win or close enough.

[p]: https://perf.link/#eyJpZCI6IngxaTZtdW1pM3g4IiwidGl0bGUiOiJmb3JFYWNoIHZzLiBTeW1ib2wuaXRlcmF0b3IgaXRlcmF0aW9uIiwiYmVmb3JlIjoiY29uc3QgYXJyYXkgPSBBcnJheS5mcm9tKHsgbGVuZ3RoOiA1MDAwMCB9LCAoXywgaSkgPT4gaSk7XG5jb25zdCBvcCA9IChpdGVtKSA9PiB7XG4gIE1hdGgucmFuZG9tKCkgKiBpdGVtO1xufTsiLCJ0ZXN0cyI6W3sibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJhcnJheS5mb3JFYWNoKG9wKTsiLCJydW5zIjpbNDUwMCw3MjUwLDgwMDAsNjAwMCw2NzUwLDY3NTAsNzUwMCw3NTAwLDYyNTAsNzI1MCw2NzUwLDgwMDAsNzAwMCw2NzUwLDg3NTAsNzAwMCw1MDAwLDU1MDAsNzUwMCwyMjUwLDU3NTAsNTc1MCw5MDAwLDcwMDAsNjI1MCw3MjUwLDY1MDAsNzI1MCw3MDAwLDY3NTAsNzAwMCw2NzUwLDYyNTAsNTUwMCw5MjUwLDY3NTAsNjc1MCw4NzUwLDY3NTAsNjI1MCw5NTAwLDQ1MDAsNzc1MCw2MjUwLDYyNTAsNjc1MCw4NTAwLDc1MDAsNTUwMCw1MjUwLDY1MDAsNzI1MCw2MDAwLDYwMDAsNDc1MCw2MjUwLDgwMDAsNzI1MCw3NTAwLDQyNTAsNjc1MCw2MjUwLDc3NTAsNjc1MCw1MjUwLDUwMDAsNjc1MCw3MDAwLDcyNTAsNzc1MCw4MjUwLDQ3NTAsODUwMCw2MjUwLDc1MDAsNzc1MCw2NTAwLDQwMDAsNzI1MCw2MDAwLDkwMDAsODAwMCw4NzUwLDYwMDAsMzUwMCw2NzUwLDYyNTAsNjc1MCw2NTAwLDU1MDAsNTI1MCw3NTAwLDU3NTAsNTI1MCw2NzUwLDg3NTAsODAwMCw3NzUwLDY1MDAsNjUwMF0sIm9wcyI6NjcwNX0seyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6ImZvciAobGV0IGl0ZW0gb2YgYXJyYXkpIHtcbiAgb3AoaXRlbSk7XG59IiwicnVucyI6WzIwMDAsMjc1MCwzMjUwLDI1MDAsMjc1MCwzMjUwLDI1MDAsMjc1MCwyNzUwLDMyNTAsMjAwMCwyNTAwLDMwMDAsMjUwMCwyNzUwLDI3NTAsMjc1MCwyNTAwLDI3NTAsMjc1MCwyNTAwLDMwMDAsMzAwMCwyMjUwLDI1MDAsMjc1MCwyNzUwLDMwMDAsMjc1MCwyNTAwLDMwMDAsMjc1MCwyNzUwLDMwMDAsMjAwMCwyNzUwLDI3NTAsMjI1MCwyNzUwLDI1MDAsMzAwMCwyMDAwLDMwMDAsMjUwMCwyNzUwLDI1MDAsMjc1MCwyNzUwLDI3NTAsMzI1MCwyNzUwLDMwMDAsMjUwMCwyNTAwLDMyNTAsMjI1MCwzMjUwLDMyNTAsMjUwMCwyMjUwLDI3NTAsMjc1MCwyNzUwLDI1MDAsMzAwMCwyMDAwLDI1MDAsMjI1MCwzMDAwLDI3NTAsMjc1MCwxNzUwLDMwMDAsMjUwMCwyNzUwLDMwMDAsMjc1MCwxNzUwLDI1MDAsMjAwMCwyNzUwLDI1MDAsMzAwMCwyMjUwLDE3NTAsMjUwMCwyNTAwLDI1MDAsMzI1MCwyMjUwLDI3NTAsMjUwMCwzMjUwLDI1MDAsMjc1MCwyMjUwLDE3NTAsMjc1MCwyNzUwLDI1MDBdLCJvcHMiOjI2NDJ9XSwidXBkYXRlZCI6IjIwMjItMDktMjdUMDI6NDc6MDQuNDEwWiJ9